uber/models/attendee.py

```diff
@@ -739,6 +853,64 @@ def possible_and_current(self):
         jobs.extend(self.possible)
         return sorted(jobs, key=lambda j: j.start_time)
 
+    # ========================================================================
+    # TODO: Refactor all this stuff regarding assigned_depts and
+    #       requested_depts. Maybe a @suffix_property with a setter for the
+    #       *_ids fields? The hardcoded *_labels props are also not great.
+    # ========================================================================
+
+    @classproperty
+    def extra_apply_attrs(cls):
+        return set(['assigned_depts_ids']).union(
+            cls.extra_apply_attrs_restricted)
+
+    @classproperty
+    def extra_apply_attrs_restricted(cls):
+        return set(['requested_depts_ids'])
+
+    @property
+    def assigned_depts_labels(self):
+        return [d.name for d in self.assigned_depts]
+
+    @property
+    def requested_depts_labels(self):
+        return [d.name for d in self.requested_depts]
+
+    @property
+    def assigned_depts_ids(self):
+        return [str(d.id) for d in self.assigned_depts]
+
+    @assigned_depts_ids.setter
+    def assigned_depts_ids(self, value):
+        from uber.models.department import Department
+        values = set(
+            None if s == 'None' else s for s in listify(value) if s != '')
+        for membership in list(self.dept_memberships):
+            if membership.department_id not in values:
+                # Manually remove dept_memberships to ensure the associated
+                # rows in the dept_membership_dept_role table are deleted.
+                self.dept_memberships.remove(membership)
+        self._set_relation(Department, 'assigned_depts', value)
+
+    @property
+    def requested_depts_ids(self):
+        return [str(d.department_id) for d in self.dept_membership_requests]
+
+    @requested_depts_ids.setter
+    def requested_depts_ids(self, value):
+        from uber.models.department import DeptMembershipRequest
+        values = set(
+            None if s == 'None' else s for s in listify(value) if s != '')
```

Comment:
This line could use some parenthesis around the if/else portion, since I had to read it several times before I got what was happening.  Alternatively you could split it up across two lines:

```python
values = {
    None if s == 'None' else s
    for s in listify(value) if s != ''
}
```
I think either way would help with readability.

Reply:
Slightly off-topic, but I constantly confuse python's set literal notation with dict literals. Like, my brain sees curly braces and I automatically start parsing it as a dict ☹️